dissect/target/plugins/apps/browser/chromium.py

```diff
@@ -562,14 +540,147 @@ def passwords(self, browser_name: str | None = None) -> Iterator[BrowserPassword
+
+        # Windows Google Chrome versions > 130 / 127 and Microsoft Edge >~ 130 use App Bound Protection
+        # (``os_crypt.app_bound_encrypted_key``) which can be decrypted using System DPAPI -> User DPAPI
+        # -> (optionally AES GCM if Google Chrome).
+        # Reference: https://github.com/chromium/chromium/tree/main/chrome/elevation_service
+        if b64_abe := local_state_conf["os_crypt"].get("app_bound_encrypted_key"):
+            try:
+                app_bound_encrypted_key = base64.b64decode(b64_abe)
+
+                if (header := app_bound_encrypted_key[0:4]) != b"APPB":
+                    self.target.log.warning(
+                        "Encountered unexpected app bound encrypted key in %s: Invalid header %r",
+                        local_state_path,
+                        header,
+                    )
+                    return keys
+
+                system_ciphertext = app_bound_encrypted_key[4:].strip(b"\x00")
+                user_ciphertext = self.target.dpapi.decrypt_system_blob(system_ciphertext)
+                plaintext = self.target.dpapi.decrypt_user_blob(user_ciphertext, username)
+
+                s_plaintext = c_elevation.Envelope(plaintext)
+
+                if s_plaintext.ciphertext_len == 0x20:
+                    # Microsoft Edge just stores the 32 byte decryption key
+                    self.target.log.info("Decrypted Microsoft Edge ABE key: %r", s_plaintext.ciphertext)
+                    keys.app_bound_key = s_plaintext.ciphertext
+
+                else:
+                    # Google Chrome has encrypted the AES decryption key using either AES GCM or ChaCha20 Poly1305
+                    # using a static key.
+                    data = c_elevation.GoogleChromeCipher(s_plaintext.ciphertext)
+
+                    if data.flag == 0x01:
+                        key = bytes.fromhex("b31c6e241ac846728da9c1fac4936651cffb944d143ab816276bcc6da0284787")
+                        cipher = AES.new(key=key, mode=AES.MODE_GCM, nonce=data.iv)
+
+                    elif data.flag == 0x02:
+                        key = bytes.fromhex("e98f37d7f4e1fa433d19304dc2258042090e2d1d7eea7670d41f738d08729660")
+                        cipher = ChaCha20_Poly1305.new(key=key, nonce=data.iv)
+
+                    else:
+                        raise ValueError("Unsupported ElevationService key flag {data.flag!r}")  # noqa: TRY301
+
+                    aes_key = cipher.decrypt_and_verify(data.ciphertext, data.mac_tag)
+
+                    self.target.log.info("Decrypted Google Chrome ABE key: %r", aes_key)
+                    keys.app_bound_key = aes_key
+
+            except (KeyError, ValueError, EOFError) as e:
+                self.target.log.warning(
+                    "Unable to decode Chromium os_crypt app_bound_key in %s: %s", local_state_path, e
+                )
+                self.target.log.debug("", exc_info=e)
+
+        return keys
+
+    @internal
```

Comment:
Is there a specific reason you made this `@internal` rather than just a function in this file?

Reply:
It doesn't need to be marked `@internal` for that does it? Just be a method. Or is there another reason you want this `@internal`?

For reference, `@internal` marks a method that can only be used in the Python API on a target object (`t.myinternalfunction`) but is not available through `target-query` (which requires an `@export`). If you want this to be treated as a regular class method you don't need an explicit `@internal`.